### ISSUE TITLE
Building BUILD_SHARED_LIBS is broken

### ISSUE THREAD

**Issue by joa-quim** [libssh2/libssh2#973]
Hi,

It  creates the dll but not the import lib

Win11, VisualStudio and libssh2 version 1.10.0

```
LINK: command "C:\PROGRA~1\MICROS~4\2022\COMMUN~1\VC\Tools\MSVC\1435~1.322\bin\Hostx64\x64\link.exe /nologo example\CMakeFiles\example-sftp_write_nonblock.dir\sftp_write_nonblock.c.obj /out:example\example-sftp_write_nonblock.exe /implib:example\example-sftp_write_nonblock.lib /pdb:example\example-sftp_write_nonblock.pdb /version:0.0 /machine:x64 /INCREMENTAL:NO /subsystem:console src\libssh2.lib ws2_32.lib kernel32.lib user32.lib gdi32.lib winspool.lib shell32.lib ole32.lib oleaut32.lib uuid.lib comdlg32.lib advapi32.lib /MANIFEST /MANIFESTFILE:example\example-sftp_write_nonblock.exe.manifest" failed (exit code 1181) with the following output:
LINK : fatal error LNK1181: cannot open input file 'src\libssh2.lib'
```

**Comment by vszakats** [comment]
Did you try with `BUILD_STATIC_LIB=OFF`? With MSVC, if you enable building the shared lib, but do not disable the static one, the implib is named `libssh2_imp.lib`, to avoid overwriting the static `libssh2.lib`.

**Comment by joa-quim** [comment]
Thanks for your quick suggestion.

Don't remember if I tried that particular combination but probably not because I never say a ``libssh2_imp.lib``.
But I tried to build ``libcurl`` using ``libssh2.lib`` and got tons of errors (missing symbols).

Any way, I solved the problem by building ``libssh2`` with ``vcpkg`` and next I used their, patched, source code and now I have one ``libssh2`` linked with ``openssl``, which was my goal.

You may close this issue if you whish.

**Comment by vszakats** [comment]
@joa-quim Thanks. Closing this for now, that said, your feedback is welcome with recent out-of-the box CMake builds. (in theory these should also make vcpkg patches unnecessary.)

**Comment by vszakats** [comment]
This was most likely an issue fixed in 1f0fe7443a1ecddd320f2c693607b2afee9bbe2f via #470. Before that, `dllexport` was not used when creating the DLL, thus no exported symbols and thus no import lib created by the linker. This received a few more updates later, but it should work fine now in master, also with autotools.
